### PR TITLE
feat: add instruction for unsupport env

### PR DESCRIPTION
**Pull request by yqz0203**
Address #41.

Following features are added:
1. A reusable component based on Modal to show instruction, and a hook to help reduce state logic.
2. Environment checking has been added to `connect wallet` button’s click event handler.

### CODE REVIEW

File: src/components/Web3Connect/index.tsx [hunk 1]
```diff
@@ -22,16 +37,47 @@ export default function Web3Connect({ color = 'gray', size = 'sm', className = '
...
+                    {' '}
+                    chrome metatask extension{' '}
+                  </a>
+                </p>
+              </>
+            ) : (
+              <>
+                {i18n._(t`${deviceInfo.browserName} is not supported`)} {t`please install`}{' '}
+                <a className="text-blue" href="https://metamask.io/">
+                  MetaTask App
```
Comment (kasumi-1) [inline src/components/Web3Connect/index.tsx:74]: metatask -> metamask

File: src/components/Web3Connect/index.tsx [hunk 2]
```diff
@@ -22,16 +37,47 @@ export default function Web3Connect({ color = 'gray', size = 'sm', className = '
...
+            {deviceInfo.isChrome && deviceInfo.isDesktop ? (
+              <>
+                <p>
+                  {i18n._(t`Chrome is supported, but first you need to install`)}
+                  <a
+                    className="text-blue"
+                    href="https://chrome.google.com/webstore/detail/metamask/nkbihfbeogaeaoehlefnkodbefgpgknn"
+                  >
+                    {' '}
+                    chrome metatask extension{' '}
```
Comment (kasumi-1) [inline src/components/Web3Connect/index.tsx:66]: metatask -> metamask

File: src/components/Web3Connect/index.tsx [hunk 3]
```diff
@@ -22,16 +37,47 @@ export default function Web3Connect({ color = 'gray', size = 'sm', className = '
...
+        {...rest}
+      >
+        {i18n._(t`Connect to a wallet`)}
+      </Button>
+      <InstructionModal
+        isOpen={isInstructionModalOpen}
+        onDismiss={onInstructionModalDismiss}
+        instruction={
+          <div>
+            {deviceInfo.isChrome && deviceInfo.isDesktop ? (
```
Comment (kasumi-1) [inline src/components/Web3Connect/index.tsx:57]: what about firefox?

File: src/components/Web3Connect/index.tsx (hunk 1 above)
Comment (kasumi-1) [reply src/components/Web3Connect/index.tsx:74]: also these should be in translation (  i18n._t(``)  )

Comment (yqz0203) [reply src/components/Web3Connect/index.tsx]: fixed.

File: src/components/Web3Connect/index.tsx (hunk 2 above)
Comment (yqz0203) [reply src/components/Web3Connect/index.tsx]: fixed.

File: src/components/Web3Connect/index.tsx (hunk 3 above)
Comment (yqz0203) [reply src/components/Web3Connect/index.tsx]: added firefox and edge plugin tips

**Pull request merged**